test/metabase/query_processor/util/nest_query_test.clj

```diff
@@ -1,276 +1,154 @@
 (ns metabase.query-processor.util.nest-query-test
   (:require
-   [clojure.set :as set]
    [clojure.test :refer :all]
-   [clojure.walk :as walk]
    [metabase.driver :as driver]
    [metabase.lib.core :as lib]
-   [metabase.lib.metadata :as lib.metadata]
    [metabase.lib.test-metadata :as meta]
    [metabase.lib.test-util :as lib.tu]
    [metabase.lib.test-util.macros :as lib.tu.macros]
    [metabase.query-processor :as qp]
    [metabase.query-processor.preprocess :as qp.preprocess]
-   ^{:clj-kondo/ignore [:deprecated-namespace]} [metabase.query-processor.store :as qp.store]
    [metabase.query-processor.util.add-alias-info :as add]
    [metabase.query-processor.util.nest-query :as nest-query]
-   [metabase.test :as mt]
-   [metabase.util :as u]))
+   [metabase.test :as mt]))
 
-;;; TODO (Cam 7/18/25) -- update all the tests that use `with-temp` to use mock metadata providers instead.
-
-;; TODO -- this is duplicated with [[metabase.query-processor.util.add-alias-info-test/remove-source-metadata]]
-(defn- remove-source-metadata [x]
-  (walk/postwalk
-   (fn [x]
-     (if ((every-pred map? :source-metadata) x)
-       (dissoc x :source-metadata)
-       x))
-   x))
-
-(defn- nest-expressions [query]
+(defn- nest-expressions-mbql5 [query]
   (driver/with-driver (or driver/*driver* :h2)
     (-> query
         qp.preprocess/preprocess
-        lib/->legacy-MBQL
-        :query
-        nest-query/nest-expressions
-        remove-source-metadata)))
+        nest-query/nest-expressions)))
+
+(defn- nest-expressions [query]
+  (-> query
+      nest-expressions-mbql5
+      lib/->legacy-MBQL
+      :query))
 
 (deftest ^:parallel nest-expressions-test
-  (driver/with-driver :h2
-    (qp.store/with-metadata-provider meta/metadata-provider
-      (is (=? (lib.tu.macros/$ids venues
-                {:source-query {:source-table $$venues
-                                :expressions  {"double_price" [:* [:field %price {::add/source-table  $$venues
```

Comment:
All the add-alias-info aliases here are really irrelevant to testing this query transformation... I think a really old version of this code attempted to update this info as part of its work but a long time ago I decided that was crazy and just had it run `add-alias-info` a second time to recalculate stuff. Since then it's just been noise on the test output, so I removed it.